Sources/App/GithubService+IssueComment.swift

```diff
@@ -0,0 +1,89 @@
+import Vapor
+import Stevenson
+
+struct PingAction: Content {
+    let zen: String
+}
+
+struct CommentAction: Content {
+    let action: String
+    let comment: Comment
+    let issue: Issue
+
+    struct Comment: Content {
+        let body: String
+    }
+
+    struct Issue: Content {
+        let number: Int
+    }
+
+    struct PullRequest: Content {
+        let url: URL
+    }
+}
+
+extension GitHubService {
+    func webhook<T: Decodable>(from request: Request) throws -> Future<T> {
+        let headers = request.http.headers
+
+        guard
+            headers.firstValue(name: .userAgent)?.hasPrefix("GitHub-Hookshot/") == true,
+            headers.firstValue(name: .init("X-GitHub-Event")) == "issue_comment"
```

Comment:
I'm guessing the header they use is the same for _Issue_ comment and _PR_ comment? (might be worth a quick comment in code if so)

```suggestion
            headers.firstValue(name: .init("X-GitHub-Event")) == "issue_comment" // Note: same used for both issue and PR comments
```

Reply:
I think it's better to link to the doc. there is only this kind of events, all pull requests are issues  in github apis AFAIK

Reply:
Ok, that's what I suspected. Link to the doc would indeed be better then 👍